entities.py

```diff
@@ -0,0 +1,315 @@
+"""
+To-do:
+
+Sprites for different entities
+
+
+Notes:
+
+Removed need for global variables. Now initializing player requires x, y, and grid, and each monster needs x, y, player and grid.
+It's a bit more of a pain to initialize (more variables required) but we can change stuff and feed it in, instead of them grabbing the global 
+    value for 'player' and 'grid'.
+"""
+
+from random import randint
+
+
+
+"""Changes:"""
+
+# added 'effect' -> basically, being poisoned or something. It's a dictionary, with True or False for whether the effect is active. We can add in what the effect actually does eventually. 
+# also added the action of being effected, and can call Entity.active_effects() to get a list of all active effects. 
+
+# added in items, and made it so items can be added to the inventory and used, along with some fun text descriptions for their appearence and usage.
+# made potion class, for easy potion making (writing all those descriptions is a pain, this way you just have to put what they do)
+
+# also added max health, so that when the player uses potions they can't go past their max health. 
+
+# ex: Anna.effected('poisoned') -> "You've been poisoned!" -> Anna.effect['poisoned'] = True -> Anna.active_effects = ['poisoned'] -> cure_potion = Potion('cure','poisoned') 
+# 	  cure_potion.pickup(Anna) -> (fun flair text) -> Anna.inventory = {'Weird Blue Potion' , 1} -> cure_potion.use(Anna) -> (fun cured flair text) -> Anna.active_effects = []
+
+
+class Entity(object):
+    def __init__(self, grid, direction="U", health=30, maxhealth=30, speed=1, accuracy=2, flatDamage = 2, damageRange=2, damageMod=2, vision=3, armor=10, phasing = False, name = None, effect = dict()):
+        # self.xpos = xpos
+        # self.ypos = ypos
+        self.grid = grid       
+        self.direction = direction #direction can be U for up, D for down, L for left, R for right
+        self.speed = speed
+        self.health = health
+        self.maxhealth = maxhealth
+        self.effect = effect
+        self.accuracy = accuracy
+        self.flatDamage = flatDamage
+        self.damageRange = damageRange
+        self.damageMod = damageMod
+        self.vision = vision #sight radius
+        self.armor = armor
+        self.phasing = phasing
+        self.directionCoordinates = {"U":(0,-1),"D":(0,1),"L":(-1,0),"R":(1,0)} # a table of which directions means which coordinates
+        print self.directionCoordinates
+    def attackRoll(self): #1d20+accuracy, if it exceeds armor class it's a hit
+        return randint(1,20)+self.accuracy #roll a 20-sided dice and add accuracy to the roll - average is 10.5 + accuracy
```

Comment:
Personally, I like having an attack roll. Maybe display damage done so that even if the player 'misses', they can see that they were trying to do the right thing.